charmcraft/charm_builder.py

```diff
@@ -188,7 +188,8 @@ def handle_dispatcher(self, linked_entrypoint):
             )
             with dispatch_path.open("wt", encoding="utf8") as fh:
                 fh.write(dispatch_content)
-                make_executable(fh)
+                if os.name == "posix":
```

Comment:
Isn't this always the case?

In any case, we can NOT do this: if we build a charm without an executable `dispatch`, it's a broken charm.

Reply:
In a future world of Windows support, the dispatch file would likely be an `.exe` file, we just wouldn't need to set an executable bit.

Reply:
(and in that case, os.name would be "nt")